### PR TITLE
Organizes the code

### PR DESCRIPTION
**Pull request by andremw**
Changes:

**addCodeTag**
Renamed addCodeTag to addCodeTagToElements and now it returns an array with the created `<code>` elements. This is good because when we change the code to highlight only a few elements (instead of calling `Prism.highlightAll()`) we'll already have the elements we need to pass to `Prism.highlightElement()`.

**waitForRender**:
Now it returns a promise. This makes it easier and clearer to work with async code.
Changed it to receive a selector as parameter for better reuse.

### CODE REVIEW

File: extension/app/highlighter.js [hunk 1]
```diff
@@ -1,36 +1,48 @@
...
     'use strict';
 
     const INTERVAL = 50; // Interval in milliseconds.
 
+    return {
+        init() {
+            insertStyles();
+            waitForRender('.diff-container').then(() => {
+                prepareLanguageClasses();
+                addCodeTagToElements('.source');
```
Comment (rproenca) [inline extension/app/highlighter.js]: As this method is only used inside this module it does not make sense for me to pass ".source"' as parameter.
".source" can be a const value that should be available for all methods, like a classes variable.

The rest of the code is very good!

Comment (andremw) [inline extension/app/highlighter.js]: I forgot to explain above, but the reason I changed it to pass '.source' is that later we can pass '.source:not([class*="language"])', to create the `<code>` elements for the elements that are not highlighted yet.

Comment (rproenca) [inline extension/app/highlighter.js]: Ok, I got your point.

Please go ahead and merge the code :)

**Comment by andremw** [comment]
Conflicts fixed! I'll merge the code.

**Pull request merged**